backend/de.metas.manufacturing/src/main/java/org/eevolution/productioncandidate/service/PPOrderCandidateService.java

```diff
@@ -498,4 +503,35 @@ public void setDateStartSchedule(
 			ppOrderCandidateDAO.save(candidate);
 		}
 	}
+
+	public void resetByPPOrderId(@NonNull final PPOrderCandidateResetRequest request)
+	{
+		final ImmutableList<I_PP_Order_Candidate> candidates = ppOrderCandidateDAO.getByOrderId(request.getPpOrderId());
+		Quantity remainingQtyToDistribute = request.getQtyToReset();
+		for (final I_PP_Order_Candidate candidate : candidates)
+		{
+			remainingQtyToDistribute = resetPPOrderCandidate(candidate, remainingQtyToDistribute);
+		}
+		resetAllocationQtys(request.getPpOrderId());
+	}
+
+	private void resetAllocationQtys(final @NonNull PPOrderId ppOrderId)
+	{
+		final ImmutableList<I_PP_OrderCandidate_PP_Order> ppOrderAllocations = ppOrderDAO.getPPOrderAllocations(ppOrderId);
+		ppOrderAllocations.forEach(ppOrderAllocation -> ppOrderAllocation.setQtyEntered(BigDecimal.ZERO));
+
+		saveAll(ppOrderAllocations);
+	}
+
+	private Quantity resetPPOrderCandidate(@NonNull final I_PP_Order_Candidate candidate, @NonNull final Quantity remainingQtyToDistribute)
+	{
+		final BigDecimal qtyToProcess = remainingQtyToDistribute.toBigDecimal().min(candidate.getQtyProcessed());
+		candidate.setQtyToProcess(candidate.getQtyToProcess().add(qtyToProcess));
+		candidate.setQtyProcessed(candidate.getQtyEntered().subtract(candidate.getQtyToProcess()));
+		candidate.setProcessed(false);
```

Comment:
same thing for setProcessed

Reply:
changed to use QtyToProcess > 0 as a condition for setProcessed.